llvm/test/CodeGen/PowerPC/warn-stack-size.ll

```diff
@@ -0,0 +1,16 @@
+; RUN: llc -mtriple=powerpc64-unknown-unknown < %s 2>&1 | FileCheck %s
+; RUN: llc -mtriple=powerpc64le-unknown-unknown < %s 2>&1 | FileCheck %s
+
+; CHECK-NOT: warning: {{.*}} stack frame size ({{.*}}) exceeds limit (4294967295) in function 'large_stack'
+define i8* @large_stack() {
+  %s = alloca [281474976710656 x i8], align 1
+  %e = getelementptr i8, i8* %s, i64 0
```

Comment:
use opaque pointers `ptr`

Does it really work for a larger-than-4G limit? Do you not hit any other limit in LLVM codegen that might assume uint32_t or a smaller limit? You probably want to try more more stack operations.

Reply:
On PPC64, larger-than-4G stack was supported after https://github.com/llvm/llvm-project/commit/5018a5dcbe70a9e2e39341762ec0ca0ad76d28b4.

Reply:
And we have some internal users using stack larger than 4G stack.